### PR TITLE
Add 1.14 to CI

### PR DESCRIPTION
**Pull request by dturn**
**What are you trying to accomplish with this PR?**
Support the latest k8s release.

**How is this accomplished?**
Adding it to ci

**What could go wrong?**
Not much, its just CI instructions

### CODE REVIEW

**Comment by dturn** [comment]
This is unlikely to pass 1.14 CI w/o https://github.com/Shopify/kubernetes-deploy/pull/460

**Comment by KnVerey** [comment]
True, but it is currently failing during setup, so looks like we have other work to do

**Comment by dturn** [comment]
> True, but it is currently failing during setup, so looks like we have other work to do

Yep, was hoping it would just work, but not at all surprised.

**Comment by benlangfeld** [comment]
Would you mind pasting the other failures here?

**Comment by dturn** [comment]
> Would you mind pasting the other failures here?

#460 was merged. This PR is just to enable 1.14 runs on our internal CI. The failures aren't related to the code in this repo, we just haven't added support for 1.14 to our CI. I'm trying to find some time to look at this, but I don't think it's blocking anything, we just don't have as much coverage as we'd like for now.

**Comment by RyanBrushett** [comment]
- Do we want to deprecate 1.10? I acted as though we did but, checking. I can also do it in a different PR if that's better.
- I'll add a changelog entry tomorrow.
CI will still fail until probably tomorrow. Will update here then.

**Comment by dturn** [comment]
> Do we want to deprecate 1.10? I acted as though we did but, checking. I can also do it in a different PR if that's better.

I think we actually need to keep running 1.10 for a bit longer.

**Comment by RyanBrushett** [comment]
I'll do that in a separate PR. Cleaning this up shortly.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 ## next
 
+*Enhancements*
+- Officially support Kubernetes 1.14 [#461](https://github.com/Shopify/kubernetes-deploy/pull/461)
```
Comment (KnVerey) [inline CHANGELOG.md]: This should also be mentioned in the version compatibility table in the readme

Comment (dturn) [reply CHANGELOG.md]: I think we only track EoL versions in that table `Last officially supported in gem version`. Though we don't have an entry for 1.9 :(

Comment (KnVerey) [reply CHANGELOG.md]: Ah, you're right, my mistake.

**Pull request merged**